src/pages/index.tsx

```diff
@@ -24,6 +26,23 @@ const cardData: CardData[] = [
 ];
 
 const Home: NextPage = () => {
+  const { status } = useSession();
+  const router = useRouter();
+
+  if (status === "loading") {
+    return (
+      <div className="flex h-screen w-screen items-center justify-center">
+        <p>Loading...</p>
+      </div>
+    );
+  }
+
+  if (status === "unauthenticated") {
+    router.replace("/api/auth/signin");
+
+    return <></>;
+  }
+
```

Comment:
Just a suggestion:
make this part a hook, something like this:
```
export const useAuthRedirect = () => {
  const { status } = useSession();
  const router = useRouter();

  const getComponentBasedOnStatus = () => {
    if (status === "loading") {
      return (
        <div className="flex h-screen w-screen items-center justify-center">
          <p>Loading...</p>
        </div>
      );
    }

    if (status === "unauthenticated") {
      router.replace("/api/auth/signin");
      return <></>;
    }
    
    return null;
  };

  return { status, getComponentBasedOnStatus };
};
```
and use it like this: 

```
const { status, getComponentBasedOnStatus } = useAuthRedirect();

  const authComponent = getComponentBasedOnStatus();
  if (authComponent) {
    return authComponent;
  }
```


in case we want to use it in different locations, what do you think?

Reply:
Great suggestion, @zeyad-alasaad! 🎉👏

I totally agree with you on that, and I can see how much easier this will make things for us in the future! 💪🔮 Especially if we're planning on using the auth middleware to manage states based on roles and flow IDs! 🙌💻🔄🔐